lib/board.rb

```diff
@@ -20,14 +20,14 @@ def display
     "
   end
 
-  def validate_position(my_board, &get_input)
+  def validate_position(my_board, get_input, puts_provide, puts_error)
     validated = false
     until validated
-      puts 'Provide a valid position between 1 and 9 and not taken'
+      puts_provide.call
       begin
         pos = Integer(get_input.call)
       rescue StandardError
-        puts 'Provide a valid integer'
+        puts_error.call
```

Comment:
Ok, **technically**, now there are "no" `puts` calls outside the `main.rb` file. However, this workaround does not fix the structural problem posed by @BrittanyBlake .

When the project specs ask us to avoid `puts` or `gets` outside of main.rb, that means avoiding also **calls** to them outside `main.rb`.

One way to fix this is by refactoring your code.

Instead of calling external methods from inside of `board.rb`, try using `board.rb` **only** for *returning* values, which will then be output by `main.rb` using puts directly from `main.rb` and not indirectly from inside of `board.rb`.

The idea here is for you to demonstrate mastery in the separation of responsabilities of different classes, and proper code modulation.